File: src/crowstream/support/post/typeDefs.js
export const postTypeDef = `
    input PostInput {
        user_id: String!
        description: String!
        comments: [CommentInput]
        files: [String]
    }
    input CommentInput {
        user_id: String!
        description: String!
        files: [String]
    }
    type Post {
        _id: ID!
        user_id: String!
        description: String!
        comments: [Comment]
        files: [String]
    }
    type Comment {
        _id: ID!
        user_id: String!
        description: String!
        files: [String]
    }
`;

export const postQueries = `
    retrieveAllPost: [Post]!
    retrievePostByID(id_post: ID!): Post
`;

export const postMutations = `
    createPost(post: PostInput!): Post!
    updatePost(id_post: ID!, post: PostInput!): Post!
    deletePost(id_post: ID!): Boolean
`;

File: api_gateway/schema/support/post/type_defs.py
"""Post Type Defs"""

# Utilities
import graphene


class CommentInput(graphene.InputObjectType):
    """Comment GraphQL Input"""
    user_id = graphene.NonNull(graphene.String, name='user_id')
    user_nick = graphene.String(name='user_nick')
    description = graphene.NonNull(graphene.String, name='description')
    files = graphene.List(graphene.String, name='files')


class PostInput(graphene.InputObjectType):
    """Post GraphQL Input"""
    user_id = graphene.NonNull(graphene.String, name='user_id')
    user_nick = graphene.String(name='user_nick')
    title = graphene.NonNull(graphene.String, name='title')
    description = graphene.NonNull(graphene.String, name='description')
    comments = graphene.List(CommentInput, name='comments')
    files = graphene.List(graphene.String, name='files')
    

class Comment(graphene.ObjectType):
    """Comment GraphQL Type"""
    _id = graphene.NonNull(graphene.ID, name='_id')
    user_id = graphene.NonNull(graphene.String, name='user_id')
    user_nick = graphene.String(name='user_nick')
    description = graphene.NonNull(graphene.String, name='description')
    files = graphene.List(graphene.String, name='files')


class Post(graphene.ObjectType):
    """Post GraphQL Type"""
    _id = graphene.NonNull(graphene.ID, name='_id')
    user_id = graphene.NonNull(graphene.String, name='user_id')
    user_nick = graphene.String(name='user_nick')
    title = graphene.NonNull(graphene.String, name='title')
    description = graphene.NonNull(graphene.String, name='description')
    comments = graphene.List(Comment, name='comments')
    files = graphene.List(graphene.String, name='files')

    